### DISCUSSION TITLE
Failed Freshness Checks Clarification

### DISCUSSION THREAD

**Discussion by dnielleyp** [dagster-io/dagster#25249] [Q&A] [answered]
Hi team, after upgrading to 1.8.10, we've implemented freshness checks and would like to gain some clarification regarding its behaviour in the following scenario: 

1. We have a run that was launched to materialise `dbt_asset_a` and `dbt_asset_b`. Both `dbt_asset_a` and `dbt_asset_b` have freshness checks defined and both are part of `dbt_assets`. 
2. `dbt_asset_b` materialised successfully, but `dbt_asset_a` failed, causing the run to fail. 

`dbt_asset_b`'s freshness check is marked as failed due to `Execution failed`. In the run, we observed error events as such: `Dependencies for step freshness_check_39daaf4c failed: ['dbt_assets']. Not executing.` (Please refer to the attached screenshot)

Shouldn't `dbt_asset_b`'s freshness check be marked as skipped, instead of failed? 

![image](https://github.com/user-attachments/assets/921baff1-93a9-442d-8a96-3e66820fbdd6)

**Comment by bolinzzz** [comment]
We encountered the same issue. How can we rerun all the skipped freshness checks (which have been marked as failed)?
If we don't clear them, the freshness checks sensor wouldn't launch new executions moving forward. 
<img width="1495" alt="image" src="https://github.com/user-attachments/assets/970e7424-bd90-4373-9540-995ae4b1eb09">

**Comment by askvinni** [comment, marked as answer]
I came across this a few days ago and had to build a bit of a workaround, though I did briefly consider opening a PR with the "fix". I'm using quotes because I could see an argument being made for this being the desired behavior.

Long story short, the reason the check isn't re-triggered by the sensor is because it is still considered to be "planned" (see [this code block](https://github.com/dagster-io/dagster/blob/1.8.11/python_modules/dagster/dagster/_core/definitions/asset_check_factories/freshness_checks/sensor.py#L165)). The sensor only triggers for checks once the "planned" state is resolved to either a succeeded or failed state.

Now, the argument why this could be the desired behavior is to stop constantly triggering jobs that keep crashing in case there's a problem on the check itself. In our case, we just adapted the sensor slightly to still have it launch checks with an "execution failed" status.

**Comment by liwenban** [comment]
We encountered the same issue.
